File: 1305-all-elements-in-two-binary-search-trees/1305-all-elements-in-two-binary-search-trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void DFS(TreeNode* root, vector<int> &result)
    {
        if (!root)
            return ;
        result.push_back(root->val);
        DFS(root->left, result);
        DFS(root->right, result);
    }
    vector<int> getAllElements(TreeNode* root1, TreeNode* root2) {
        vector<int> result;
        DFS(root1, result);
        DFS(root2, result);
        sort(begin(result), end(result));
        return (result);
    }
};

File: 1305-all-elements-in-two-binary-search-trees/1305-all-elements-in-two-binary-search-trees.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {number[]}
 */

var travelIntoTree = function(root, arr) {
    if (root === null)
        return ;
    arr.push(root.val);
    travelIntoTree(root.left, arr);
    travelIntoTree(root.right, arr);
    
}

var getAllElements = function(root1, root2) {
    let arr = [];
    travelIntoTree(root1, arr);
    travelIntoTree(root2, arr);
    return (arr.sort((a, b) => a-b));
};